src/Data/ContentSearchResult/ContentSearchResultDataClass.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  currentLanguage,
+  extractText,
+  load,
+  Obj,
+  provideDataClass,
+  urlFor,
+} from 'scrivito'
+import { objTitle } from '../../utils/objTitle'
+import { ensureString } from '../../utils/ensureString'
+import { isImage } from '../../Objs/Image/ImageObjClass'
+
+const BLACKLIST_OBJ_CLASSES = ['Image', 'Redirect', 'Video']
+
+export const ContentSearchResult = provideDataClass('ContentSearchResult', {
+  attributes: async () => {
+    const lang = await load(currentLanguage)
+
+    return {
+      image: [
+        'reference',
+        { title: lang === 'de' ? 'Bild' : 'Image', to: 'Image' },
+      ],
+      snippet: ['string', { title: lang === 'de' ? 'Schnipsel' : 'Snippet' }],
+      title: ['string', { title: lang === 'de' ? 'Titel' : 'Title' }],
+      url: ['string', { title: 'URL' }],
+    }
+  },
+  title: async () =>
+    (await load(currentLanguage)) === 'de'
+      ? 'Inhalts-Suchergebnis'
```

Comment:
* [ ] FYI To make this more readable maybe it's sufficient to have just "search results" and (later on) "data search results". Alternatively maybe sth like "Suchtreffer (Content)"?